src/stirling/utils/linux_headers.cc

```diff
@@ -229,7 +229,20 @@ StatusOr<std::filesystem::path> ResolvePossibleSymlinkToHostPath(const std::file
     return error::Internal(ec.message());
   }
 
-  const auto resolved_host_path = system::Config::GetInstance().ToHostPath(resolved);
+  // Relative paths containing "../" can result in an invalid host mount path when using
+  // ToHostPath. Therefore, we need to treat the absolute and relative cases differently.
+  std::filesystem::path resolved_host_path = p.parent_path();
```

Comment:
You can probably push the `p.parent_path()` into the else clause since that's the only branch it's relevant for.

Reply:
Updated in ed7b839d630c0ee2acdde1319a008d6d099e0e18